rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -1091,6 +1133,11 @@ extern "C" rmw_node_t * rmw_create_node(
     return nullptr;
   }
 
+  ret = context->impl->init(&context->options);
+  if (RMW_RET_OK != ret) {
+    return nullptr;
```

Comment:
It used to clean up on failure via the destructor, but that no longer happens. Calling `fini` here would have exactly the same effect, I think.

Reply:
I think that functions that fail shouldn't leak (i.e.: if `init` failed, there shouldn't be need to explicitly call `fini`).
I used that approach in https://github.com/ros2/rmw_cyclonedds/pull/176/commits/34c9a437c9123f89ac6b7df609b4efc7c37f141a.